tests/unit/logic/vote.spec.js

```diff
@@ -0,0 +1,1012 @@
+
+				beforeEach(function () {
+					trs = {asset: {votes: 123}}
+					getBytes = function () {
+						vote.getBytes(trs)
+					}
+				})
+
+				it('should throws an error', function () {
+					expect(getBytes).to.throw()
+				})
+			})
+
+			context('if trs.asset.votes is undefined', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: undefined}}
+					result = vote.getBytes(trs)
+				})
+
+				it('should return null', function () {
+					expect(result).to.equals(null)
+				})
+			})
+
+			context('if everything is ok', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: ['+37b6d3fc89fb418fa5e3799e760cc2b1733a567daebb08a531d3d7e8b24f2d22','+9f59d4260dcd848f71d17824f53df31f3dfb87542042590554419ff40542c55e']}}
+					result = vote.getBytes(trs)
+				})
+
+				it('should return a Buffer object', function () {
+					expect(result).to.be.an.instanceof(Buffer)
+				})
+			})
+		})
+
+		describe('apply()', function () {
+			var clock, trs, block, sender, callback, scope, delegates, rounds, system
+
+			beforeEach(function () {
+				callback = sinon.spy()
+				clock = sinon.useFakeTimers()
+				scope = { account: { merge: function (address, data, cb) { cb() } } }
+				rounds = {calc: function (height) { return height }}
+				trs = {asset: {votes: []}}
+				sender = {address: 123}
+				block = {id: 123, height: 123}
+				Vote.__set__('setImmediate', setImmediate)
+				Vote.__set__('self.checkConfirmedDelegates', function (trs, cb) {
+					return setImmediate(cb)
+				})
+				Vote.__set__('self.scope', scope)
+				vote.bind(delegates, rounds, system)
+				vote.apply(trs, block, sender, callback)
+			})
+
+			afterEach(function () {
+				clock.restore()
+			})
+
+			it('should call to callback', function () {
+
+				clock.runAll()
+				expect(callback.called).to.be.true
+			})
+		})
+
+		describe('undo()', function () {
+			var clock, trs, block, sender, callback, delegates, rounds, system, account, schema, logger
+
+			beforeEach(function () {
+				callback = sinon.spy()
+				clock = sinon.useFakeTimers()
+				schema = new zSchema()
+				vote = new Vote(logger, schema)
+				account = { merge: function () {} }
+				sinon.stub(account, 'merge').callsFake(function (address, data, cb) {
+					cb()
+				})
+				Vote.__set__('self.scope', {account: account})
+				rounds = {calc: function (height) { return height }}
+				sender = {address: 123}
+				block = {id: 123, height: 123}
+				Vote.__set__('setImmediate', setImmediate)
+
+				vote.bind(delegates, rounds, system)
+			})
+
+			afterEach(function () {
+				clock.restore()
+			})
+
+			context('if trs.asset.votes is null', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: null}}
+					vote.undo(trs, block, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
```

Comment:
Also Diff.reverse shouldn't be called.

Reply:
Applies also to the next tests

Reply:
The reason is because that this test depends on solve the next issue:

#29

When that issue is solved the test will works properly.

Reply:
About spy Diff.reverse() was patched.